src/github.com/getlantern/lantern-android/README.md

```diff
@@ -0,0 +1,232 @@
+
+
+    private Button killButton;
+    private Button startButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_main);
+
+        // Initializing application context.
+        Go.init(getApplicationContext());
+
+        killButton = (Button)findViewById(R.id.stopProxyButton);
+        startButton = (Button)findViewById(R.id.startProxyButton);
+
+        // Disabling stop button.
+        killButton.setEnabled(false);
+
+        // Enabling proxy button.
+        startButton.setEnabled(true);
+    }
+
+    public void stopProxyButtonOnClick(View v) {
+
+        Log.v("DEBUG", "Attempt to stop running proxy.");
+        try {
+            Flashlight.StopClientProxy();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        };
+
+        // Disabling stop button.
+        killButton.setEnabled(false);
+
+        // Enabling proxy button.
+        startButton.setEnabled(true);
+
+    }
+
+    public void startProxyButtonOnClick(View v) {
+        Log.v("DEBUG", "Attempt to run client proxy on :9192");
+
+        try {
+            Flashlight.RunClientProxy("0.0.0.0:9192");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        // Enabling stop button.
+        killButton.setEnabled(true);
+
+        // Disabling proxy button.
+        startButton.setEnabled(false);
+    }
+}
+```
+
+After this new activity is added the *design view* will be active, drag two
+buttons from the *Pallete* into the screen.
+
+![Adding two buttons](https://cloud.githubusercontent.com/assets/385670/5769192/d9df19cc-9cdd-11e4-90d0-b37b6d6b3a41.png)
+
+Select the first button and look for the *id* property on the Properties pane,
+set it to *startProxyButton* and name the button accordingly. Look for the
+*onClick* property and choose the *startProxyButtonOnClick* value from the drop
+down.
+
+The second button's *id* must be set to *stopProxyButton* and the *onClick* to
+*stopProxyButtonOnClick*.
+
+Finally, hit the *Run app* action under the *Run* menu and deploy it to a real
+device or to an ARM-based emulator.
+
+If everything goes OK, you'll have two buttons and you can start `flashlight`
+by touching the *startProxyButton*.
+
+![Deploy to a device](https://cloud.githubusercontent.com/assets/385670/5712899/db6ddb34-9a7b-11e4-8841-6b6b12e46c27.png)
+
+As long as the app is open, you'll be able to test the canonical example by
+finding the device's IP and sending it a special request:
+
+```
+curl -x 10.10.100.97:9192 http://www.google.com/humans.txt
+# Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.
+```
```

Comment:
I'm not able to make this work, as I don't have access to the emulator's network.

What I can do however is set up redirection by first telnetting to the emulator's console and then setting up a redirect, like this:

```bash
➜  ox.to.a.cart git:(master) ✗ telnet localhost 5556
Trying ::1...
telnet: connect to address ::1: Connection refused
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
Android Console: type 'help' for a list of commands
OK
redir add tcp:9192:9192
OK
```

Then I can

```
curl -x localhost:9192 http://www.google.com/humans.txt
```